server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobStore.java

```diff
@@ -61,6 +63,36 @@ public BlobContainer blobContainer(BlobPath path) {
         return new FsBlobContainer(this, path, f);
     }
 
+    @Override
+    public void deleteBlobsIgnoringIfNotExists(Iterator<String> blobNames) throws IOException {
+        IOException ioe = null;
+        long suppressedExceptions = 0;
+        while (blobNames.hasNext()) {
+            try {
+                Path resolve = path.resolve(blobNames.next());
```

Comment:
Hmm this resolves the blob paths again. I think `FsBlobContainer#path` already includes the base path for the repo, so `blobNames.next()` should already be an absolute path, and therefore this should be a no-op.

But then I see that we've got to convert the `String` name to a `Path` somehow, and maybe this is best. Deserves a comment at least.